snippet_parser/core.py

```diff
@@ -191,94 +193,120 @@ def extract(self, wikitext):
-                for marker in tree.cssselect('.' + CITATION_NEEDED_MARKER_CLASS):
-                    root = marker.getparent()
-                    while root is not None and root.tag not in _SNIPPET_ROOT_TAGS:
-                        root = root.getparent()
-                    if root is None:
-                        continue
-                    if root.tag in _LIST_TAGS:
-                        snippet_roots = self._html_list_to_snippets(root)
-                    else:
-                        snippet_roots = [self._make_snippet_root(root)]
-            else:
-                # Throw away the actual template, we don't need it.
-                for marker in tree.cssselect('.' + CITATION_NEEDED_MARKER_CLASS):
-                    lxml_utils.remove_element(marker)
-
-                # Keep only snippet root top-level elements within the body
-                # that have any text content (we may have created empty elements
-                # above during cleanup). This is not great as any content within,
-                # say, <blockquote> gets removed entirely, but it's good enough
-                # in most cases.
-                snippet_roots = [
-                    self._make_snippet_root(*(
-                        e for e in tree.cssselect(
-                            'body > ' + ', '.join(_SNIPPET_ROOT_TAGS))
-                        if e.text_content() and not e.text_content().isspace()))
-                ]
-
-            snippets_in_section = set()
-            for sr in snippet_roots:
-                # Some last-minute cleanup to shrink the snippet some more.
-                # Remove links and attributes, but make sure to keep the
-                # class in our marker elements, and that there is no space
-                # before it (which we need for the UI).
-                lxml.etree.strip_tags(sr, 'a')
-                markers_in_snippet = sr.cssselect(
-                    '.' + CITATION_NEEDED_MARKER_CLASS)
-                lxml.etree.strip_attributes(sr, 'id', 'class', 'style')
-                sr.attrib['class'] = SNIPPET_WRAPPER_CLASS
-                for marker in markers_in_snippet:
-                    marker.attrib['class'] = CITATION_NEEDED_MARKER_CLASS
-                    lxml_utils.strip_space_before_element(marker)
-
-                length = len(sr.text_content().strip())
-                self.stats.snippet_lengths[length] += 1
-                if minlen < length < maxlen:
-                    snippet = d(lxml.html.tostring(
-                        sr, encoding = 'utf-8', method = 'html')).strip()
-                    snippets_in_section.add(snippet)
-
-            sectitle = ''
-            if i != 0:
-                # Re-parse the section title because fast_parse is
-                # configured to ignore style tags (see above and 
-                # https://github.com/earwig/mwparserfromhell/issues/40),
-                # but we do want to remove them now with strip_code().
```

Comment:
Looks like this comment got lost when splitting this as a separate method, please restore it. I can see a couple of others that got lost I think ("Make sure we don't remove elements inside the markers", "Keep only snippet root top-level elements within the body"...)

Those comments record the hacks and trade-offs I made in the parsing logic, so they're very valuable :)

Reply:
Sorry for my negligence :no_mouth: